bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/speedtestHandler.java

```diff
@@ -0,0 +1,496 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.speedtest.internal;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.PatternSyntaxException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.speedtest.internal.dto.ResultContainer;
+import org.openhab.binding.speedtest.internal.dto.ResultsContainerServerList;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link speedtestHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Brian Homeyer - Initial contribution
+ */
+@NonNullByDefault
+public class speedtestHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(speedtestHandler.class);
+    private @Nullable speedtestConfiguration config;
```

Comment:
```suggestion
    private SpeedtestConfiguration config = new SpeedtestConfiguration();
```
Will save you checking for null, also note the change to use Capital.

Reply:
done